plugins/gin-gonic/gin/example_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2020 Datadog, Inc.
+// Copyright The OpenTelemetry Authors
```

Comment:
Traditionally I've seen this the other way around:

```
// Copyright The OpenTelemetry Authors
// Copyright 2016-2020 Datadog, Inc.
``` 

INAL, but I think this comes from the idea that the original license needs to remain unmodified if copied.

Also, [the NOTICES file](https://github.com/DataDog/dd-trace-go/blob/8fb554ff7cf694267f9077ae35e27ce4689ed8b6/NOTICE) from the originating repo needs to be included.

Is there a specific reason we are including this code instead of writing our own?

Reply:
Because the original source that @krnowak used as reference has dual license choice: either mit or al2, the [mit](https://opensource.org/licenses/MIT) was chosen because it is a lot simpler and does not require the NOTICES file: https://github.com/DataDog/dd-trace-go/pull/571#issue-364984406

As for the order, I think it is ok to have opentelemetry as the second item. This PR conversation can serve as the record for how the code was derived.

Reply:
@andrewhsu: The code this was copied from was not licensed under an MIT license. It is dual licensed as Apache2 or BSD3. If we want to use the BSD3 that license will need to be copied into this file (part of the explicit requirement to reuse code with that license type), otherwise we'll need to preserve the original Apache2 copyright (including the notices file and the header):

> 4. Redistribution. You may reproduce and distribute copies of the Work or Derivative Works thereof in any medium, with or without modifications, and in Source or Object form, provided that You meet the following conditions:
> 
>    1. You must give any other recipients of the Work or Derivative Works a copy of this License; and
>    1. You must cause any modified files to carry prominent notices stating that You changed the files; and
>    1. You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works; and
>    1. If the Work includes a "NOTICE" text file as part of its distribution, then any Derivative Works that You distribute must include a readable copy of the attribution notices contained within such NOTICE file, excluding those notices that do not pertain to any part of the Derivative Works, in at least one of the following places: within a NOTICE text file distributed as part of the Derivative Works; within the Source form or documentation, if provided along with the Derivative Works; or, within a display generated by the Derivative Works, if and wherever such third-party notices normally appear. The contents of the NOTICE file are for informational purposes only and do not modify the License. You may add Your own attribution notices within Derivative Works that You distribute, alongside or as an addendum to the NOTICE text from the Work, provided that such additional attribution notices cannot be construed as modifying the License.
> 
> You may add Your own copyright statement to Your modifications and may provide additional or different license terms and conditions for use, reproduction, or distribution of Your modifications, or for any such Derivative Works as a whole, provided Your use, reproduction, and distribution of the Work otherwise complies with the conditions stated in this License.

Reply:
@MrAlias apologies, you're right. even the PR i linked to in my original comment says BSD3. 🤦 i think i used the wrong link from the wrong tab when composing the comment too quickly. also, kids: don't do drugs.

In that case, @krnowak I recommend we adhere to the AL2 since this repo is already setup up for it and with these changes to this PR:

1. add to the top of the source file a code comment that says `// Code initially derived from https://github.com/DataDog/dd-trace-go/blob/0b8240f0a30d8339b3250ff20d888f40d99759a7/contrib/gin-gonic/gin/example_test.go` or whatever is the appropriate source link
2. add `NOTICES` file to top-level of the repo with contents from https://github.com/DataDog/dd-trace-go/blob/0b8240f0a30d8339b3250ff20d888f40d99759a7/NOTICE

Reply:
For reference, I found an example of derived source that cites original source in the kubernetes project: https://github.com/kubernetes/kubernetes/blob/b19de7f9b6a23f669335a10cb1c0e8de10324a78/pkg/credentialprovider/azure/azure_acr_helper.go#L43

Reply:
@andrewhsu: The first point is already fulfilled, I write something like "// Based on https://github.com/DataDog/dd-trace-go/blob/8fb554ff7cf694267f9077ae35e27ce4689ed8b6/contrib/gin-gonic/gin/gintrace_test.go"

Will add the NOTICES file.